client/my-sites/site-settings/disconnect-site/disconnect-survey.jsx

```diff
@@ -0,0 +1,96 @@
+/** @format */
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import Card from 'components/card';
+import CompactCard from 'components/card/compact';
+import { getSelectedSite, getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { isFreeJetpackPlan } from 'lib/products-values';
+import { isJetpackSite } from 'state/sites/selectors';
+
+class DisconnectSurvey extends Component {
+	state = {
+		reasonSelected: 'onlyNeedFree',
+		renderInitial: true,
+	};
+
+	renderFollowUp() {
+		// placeholder
+		return <Card className="disconnect-site__question">{ 'follow-up' }</Card>;
+	}
+
+	logSelection = option => {
+		this.setState( {
+			reasonSelected: option.value,
+			renderInitial: false,
+		} );
+	};
+
+	getOptions() {
+		const { translate, site } = this.props;
+
+		const options = [
+			{ value: 'tooHard', label: translate( 'It was too hard to configure Jetpack' ) },
+			{ value: 'didNotInclude', label: translate( 'This plan didn’t include what I needed' ) },
+		];
+
+		if ( ! isFreeJetpackPlan( site.plan ) ) {
+			options.push( { value: 'onlyNeedFree', label: translate( 'This plan is too expensive' ) } );
+		}
+		return options;
+	}
+
+	getSurveyQuestions( options ) {
+		const questions = [];
+		for ( let i = 0; i < options.length; i++ ) {
+			questions.push(
+				<CompactCard href="#" onClick={ this.logSelection } className="disconnect-site__survey-one">
+					{ options[ i ].label }
+				</CompactCard>
+			);
+		}
+		return questions;
```

Comment:
This is the prototypical case for `map` and I have a strong preference to either the built-in [`Array.prototype.map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map) or [lodash's `map`](https://lodash.com/docs/4.17.4#map):

```jsx
return map( options, option => (
  <CompactCard
    href="#"            // <-- Is this necessary?
    onClick={ this.logSelection }
    className="disconnect-site__survey-one"
  >
    { option.label }
  </CompactCard>
) );
```

Reply:
Aren't the `value`s missing?